src/AutoMapper/Configuration/MappingExpression.cs

```diff
@@ -18,8 +18,14 @@ public void ConvertUsing<TTypeConverter>()
         public IMappingExpression ReverseMap()
         {
             var reverseMap = new MappingExpression(new TypePair(Types.DestinationType, Types.SourceType), MemberList.Source);
-            reverseMap.MemberConfigurations.AddRange(MemberConfigurations.Select(m => m.Reverse()).Where(m => m != null));
+            reverseMap.MemberConfigurations.AddRange(MemberConfigurations.Select(m => m.Reverse()).Where(m => m != null));
             ReverseMapExpression = reverseMap;
+
+            foreach (var feature in Features)

```

Comment:
I think the collection should have a method ReverseConfiguration or smth.

Reply:
The AM itself use the `Reverse()` on ex `IPropertyMapConfiguration` (line 21 above), but can add a method on the collection to return an reversed collection that will be used. I think we should keep the name `Reverse()` (to follow the above) or add it as `CreateReverseCollection()`.

Reply:
There is already a Reverse on IEnumerable, so I thought we should avoid the confusion.